master/pkg/model/user.go

```diff
@@ -86,18 +86,33 @@ func (u FullUser) ToUser() User {
 	}
 }
 
-// ValidatePassword checks that the supplied password is correct.
-func (user User) ValidatePassword(password string) bool {
-	// If an empty password was posted, we need to check that the
-	// user is a password-less user.
-	if password == "" {
-		return !user.PasswordHash.Valid
+// SetDefaultPassword initializes default password.
+func (user User) SetDefaultPassword(password string) error {
+	passwordHash, err := bcrypt.GenerateFromPassword(
+		[]byte(password),
+		BCryptCost,
+	)
+	DefaultPassword = null.StringFrom(string(passwordHash))
+	if err != nil {
+		return err
 	}
+	return nil
+}
 
+// ValidatePassword checks that the supplied password is correct.
+func (user User) ValidatePassword(password string) bool {
 	// If the model's password is empty, then
-	// supplied password must be incorrect
+	// check for default
 	if !user.PasswordHash.Valid {
-		return false
+		// If default password is not set, then
+		// supplied password must be empty
+		if !DefaultPassword.Valid {
+			return password == ""
+		}
+		err := bcrypt.CompareHashAndPassword(
```

Comment:
Looks good but but maybe better to keep it DRY (don't repeat yourself). So around ~105 could say `hash := user.PasswordHash` and then reassign here and use `hash` everywhere else